tests/providers/test_adapter.py

```diff
@@ -600,6 +604,133 @@ def gate_matches_connectivity(gate) -> bool:
             )
         )
 
+    def test_angle_restrictions_rigetti(self):
+        """Tests that angle restrictions for native gates are enforced."""
+        circuit = QuantumCircuit(1)
+        circuit.rx(np.pi / 4, 0)
+
+        restrictions = {"rx": {0: {np.pi, -np.pi, np.pi / 2, -np.pi / 2}}}
+        with pytest.raises(ValueError):
+            to_braket(circuit, basis_gates={"rx"}, angle_restrictions=restrictions)
+
+    def test_angle_restrictions_rigetti_valid(self):
+        """Tests that allowed Rigetti angles pass validation."""
+        circuit = QuantumCircuit(1)
+        circuit.rx(np.pi / 2, 0)
+
+        restrictions = {"rx": {0: {np.pi, -np.pi, np.pi / 2, -np.pi / 2}}}
+        braket_circuit = to_braket(
+            circuit, basis_gates={"rx"}, angle_restrictions=restrictions
+        )
+        assert len(braket_circuit.instructions) == 1
+
+    def test_validate_angle_restrictions_extra_index(self):
+        """Restrictions on unused parameter indices should be ignored."""
+        _validate_angle_restrictions("rx", [0.0], {"rx": {1: {0.0}}})
+
+    def test_native_angle_restrictions_default(self):
+        """Unknown device types should return an empty restriction map."""
+        properties = GateModelSimulatorDeviceCapabilities.parse_obj(
+            {
+                "braketSchemaHeader": {
+                    "name": "braket.device_schema.simulators.gate_model_simulator_device_capabilities",
+                    "version": "1",
+                },
+                "service": {"executionWindows": [], "shotsRange": [1, 10]},
+                "action": {
+                    "braket.ir.jaqcd.program": {
+                        "actionType": "braket.ir.jaqcd.program",
+                        "version": ["1"],
+                        "supportedOperations": ["x"],
+                    }
+                },
+                "paradigm": {"qubitCount": 2},
+                "deviceParameters": {},
+            }
+        )
+        assert native_angle_restrictions(properties) == {}
```

Comment:
The linter picked this line to improve the code, and it is blocking the merging until we do its bidding. It is suggesting 
```suggestion
        not native_angle_restrictions(properties)
```